### ISSUE TITLE
Warpinator Connection Error on Linux Mint 20.0

### ISSUE THREAD

**Issue by Twente-54** [linuxmint/warpinator#104]
I am having a problem with Warpinator running on a linux mint 20.2 laptop. It cannot get it to connect with Warpinator on my Sony Xperia phone (android 10) because of a "connection error". I enabled UDP etc on gate 42000 in the firewall but that does not help. Looking at the adroid app it appears that it is trying to connect with "null@<computer name>" whereas it should be trying to connect with "<my name>@<computer name>". I can understand that trying to connect with "null@<computer name" gets it nowhere, although it does seem to mention a correct IP adress?

I have a second laptop running mint 20.0; this has no problem with the android app on the phone and connects without any problems as it allways did. Never changed anything in the firewall settings. It won't however connect with the mint 20.2 laptop. Also here it is trying to connect with "<computer name>" instead of "<my name>@<computer name>". Even worse is that the mint 20.2 Warpinator doesn't even see the mint 20.0 laptop; it is not mentioned in it's list of connectable systems.

If I disable the firewall on 20.2 then it cán connect to the android warpinator, that then is correctly stating that it is connecting with "<my name>@<computer name>". It will however still not connect with mint 20.0 for the same reason; it still does not show it in the list. Warpinator on mint 20.0 is then still trying to connect to "computer name".

**Comment by Twente-54** [comment]
I see that some text is dropped; it should read like this:

I am having a problem with Warpinator running on a linux mint 20.2 laptop. It cannot get it to connect with Warpinator on my Sony Xperia phone (android 10) because of a "connection error". I enabled UDP etc on gate 42000 in the firewall but that does not help. Looking at the adroid app it appears that it is trying to connect with "null@computer name" whereas it should be trying to connect with "my name@computer name". I can understand that trying to connect with "null@computer name" gets it nowhere, although it does seem to mention a correct IP adress?

I have a second laptop running mint 20.0; this has no problem with the android app on the phone and connects without any problems as it allways did. Never changed anything in the firewall settings. It won't however connect with the mint 20.2 laptop. Also here it is trying to connect with "computer name" instead of "my name@computer name". Even worse is that the mint 20.2 Warpinator doesn't even see the mint 20.0 laptop; it is not mentioned in it's list of connectable systems.

If I disable the firewall on 20.2 then it cán connect to the android warpinator, that then is correctly stating that it is connecting with "my name@computer name". It will however still not connect with mint 20.0 for the same reason; it still does not show it in the list. Warpinator on mint 20.0 is then still trying to connect to "computer name".

**Comment by mtwebster** [comment]
Hi, can you run warpinator in debug mode on the laptops:
```
warpinator --debug
```
and paste the output (or [dpaste](https://dpaste.org/)) these while they try to connect. Please keep the android device inactive (or turn off wifi) for this, so only the connections between the two laptops are logged.

**Comment by Twente-54** [comment]
Hi Michael,

As per your request. I added a screenprint of both windows. They are in
Dutch; the 20.2 one says as much as that it can't find another computer.
I also added screen prints of the firewall settings.
Succes tracing the problem; I hope it wasn't me doing something silly.

Kind regards
Jan Stegehuis (Twente-54)



Op zo 19 sep. 2021 om 18:38 schreef Michael Webster <
***@***.***>:

**Comment by mtwebster** [comment]
Hi I need you to run warpinator --debug on both laptops so I can gather information - also, there are no screenshots attached here (you should be able to drag the files from the file manager into the response box here).

**Comment by Twente-54** [comment]
I did run warpinator in debug mode.

I answered your email and added the debug results and screenshots as
attachments.

[image: image.png]

If that is not the way to go about this, then tell me what is.





Op ma 20 sep. 2021 om 15:36 schreef Michael Webster <
***@***.***>:

**Comment by Twente-54** [comment]
Trying it in github

Succes

[debug warpinator mint 20-0.txt](https://github.com/linuxmint/warpinator/files/7196170/debug.warpinator.mint.20-0.txt)
[debug warpinator mint 20-2.txt](https://github.com/linuxmint/warpinator/files/7196171/debug.warpinator.mint.20-2.txt)
![screen print fire wall settings mint 20-0](https://user-images.githubusercontent.com/91020169/134013762-8615ceb1-0775-4b2e-8fcc-d179c1342b8c.png)
![screen print fire wall settings mint 20-2](https://user-images.githubusercontent.com/91020169/134013764-92212afd-1fec-4a38-930d-73d4aaf2974c.png)
![screen print warpinator mint 20-0](https://user-images.githubusercontent.com/91020169/134013767-abbebeb1-a599-4c93-babe-c62895f04b62.png)
![screen print warpinator mint 20-2](https://user-images.githubusercontent.com/91020169/134013769-909ab929-8c16-4bbf-91fb-26dfeb69bc7c.png)
.

**Comment by mtwebster** [comment]
So both of those pcs are using the most recent (or close to it) version. For those to communicate with one another they each need to have tcp ports 42000 and 42001 open.

You need to also open 42000 udp on both, which should allow backward compatibility to connect to the android device. _I noticed you opened udp port 4200 rather than 42000 on the one machine._

So, to summarize, both pcs should have the following open:
42000 tcp
42001 tcp
42000 udp

Try fixing these things and we'll see where to go from there.

**Comment by Twente-54** [comment]
I fixed the firewall setting according to your specs and the laptops can now communicate both with each other and with the android telephone.

I added the UDP setting to the mint 20-2 firewall by removing the "/tcp" from the 42000 entry; it now looks like this.
![screen print fire wall settings mint 20-2](https://user-images.githubusercontent.com/91020169/134969786-0cce93a5-da3c-4f59-b7d4-ac3f3ff152c1.png)
To do this I had to remove the original 42000/tcp entry that Warpinator had added to the rules when being installed en add a new entry since such entries can't be changed.

I added the 42001/tcp entry to the mint 20-2 firewall settings since it was missing there; it now looks the same as the mint 20-2 settings. The Mint 20-0 Warpinator does not add this entry when it is installed. It does however correctly add the 42000 entry without the "/tcp" part.

As I said: " I hope it wasn't me doing something silly." and it turns out it wás me. Well to the larger extent.
Thanks for your help.

Regards Jan (Twente-54)
